### DISCUSSION TITLE
Improve debugging: How to inspect execution time of function easily in rust and node

### DISCUSSION THREAD

**Discussion by hyfdev** [rolldown/rolldown#876] [Core Minutes]
Met a lot of cases that want to simply check how long a function run.

We now could do this manually by adding hard code to print out the time. While it works, it's not friendly to be used.

In rust:

We are already using the `tracing` crate. Might able to do something with it.

In node

Well, I'm currently blind.

**Comment by milesj** [comment]
I've wanted to give this a try https://crates.io/crates/tracing-chrome

**Reply by hyfdev** [reply]
We are using it right now and it's indeed quite useful. But I still feel it's too heavy for checking out execution of single function. https://github.com/rolldown/rolldown/blob/f38c27bc0549fd4dba2011e03df3abb88bd85797/crates/rolldown_tracing/src/lib.rs#L32-L46

It needs you to open browser and select files, select directories, and parsing, clicking, scaling and scrolling. :)

**Reply by hyfdev** [reply]
What I imagine is we could simply run `RUST_LOG=[some filter syntax] cargo run ....` or `RUST_LOG=[some filter syntax]  pnpm build` and the logging would be

```
calling fn xxxx() takes xxx ms
```

we just do nothing but add `#[instrument]` on that function.
